packages/api/amplify_api/lib/src/graphql/utils.dart

```diff
@@ -51,8 +51,8 @@ _RelatedFields _getRelatedFields(ModelSchema modelSchema) {
   return _fieldsMemo[modelSchema]!;
 }
 
-ModelField? getBelongsToFieldFromModelSchema(ModelSchema modelSchema) {
-  return _getRelatedFields(modelSchema).singleFields.firstWhereOrNull((entry) =>
+Iterable<ModelField> getBelongsToFieldFromModelSchema(ModelSchema modelSchema) {
```

Comment:
nit: could you please rename this `getBelongsToFieldsFromModelSchema` so it's clear it returns multiple.